x-pack/osquerybeat/beater/action_handler.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package beater
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/internal/config"
+)
+
+type actionHandler struct {
+	inputType string
+	bt        *osquerybeat
+}
+
+func (a *actionHandler) Name() string {
```

Comment:
requires pointer?  not important, just curious

Reply:
it is passed into the the agent client as interface so will escape to the heap no matter what, and it's only allocated once